### PR TITLE
Skip postcss config location resolving in node_modules

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Skip postcss config resolving in node_modules for Turbopack. This matches current Next.js on webpack closer as that only resolves the postcss config from the project root. The additional feature Turbopack has is that it can find a postcss config relative to the css being compiled. This is useful for e.g. reducing the amount of overhead certain postcss plugins that only have to run against a single file take (i.e. Tailwindcss). 

Depends on github.com/vercel/turbo/pull/7012 landing first.

Closes NEXT-2109

### CODE REVIEW

**Comment by timneutkens** [comment]
* `canary`
  * **#60697** <a href="https://app.graphite.dev/github/pr/vercel/next.js/60697?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/next-swc/crates/next-core/src/next_server/context.rs [hunk 1]
```diff
@@ -648,14 +674,14 @@ pub async fn get_server_module_options_context(
...
-                enable_postcss_transform: postcss_transform_options,
+                enable_postcss_transform,
                 enable_typescript_transform: Some(tsconfig),
                 enable_mdx_rs,
                 decorators: Some(decorators_options),
                 rules: vec![
                     (
                         foreign_code_context_condition,
-                        module_options_context.clone().cell(),
+                        foreign_code_module_options_context.cell(),
```
Comment (sokra) [inline packages/next-swc/crates/next-core/src/next_server/context.rs:684]: That was a bug before too

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| buildDuration | 13s | 12.9s | N/A |
| buildDurationCached | 7.1s | 6.3s | N/A |
| nodeModulesSize | 200 MB | 200 MB | N/A |
| nextStartRea..uration (ms) | 435ms | 435ms | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| 193.HASH.js gzip | 181 B | 182 B | N/A |
| 3f784ff6-HASH.js gzip | 53.4 kB | 53.4 kB | ✓ |
| 433-HASH.js gzip | 28.9 kB | 28.9 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 240 B | 244 B | N/A |
| main-HASH.js gzip | 31.8 kB | 31.8 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 98.6 kB | 98.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 183 B | 181 B | N/A |
| amp-HASH.js gzip | 504 B | 502 B | N/A |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 253 B | N/A |
| head-HASH.js gzip | 350 B | 349 B | N/A |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.28 kB | 4.28 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.61 kB | ✓ |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.4 kB | 3.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| index.html gzip | 528 B | 528 B | ✓ |
| link.html gzip | 541 B | 540 B | N/A |
| withRouter.html gzip | 524 B | 522 B | N/A |
| Overall change | 528 B | 528 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.9 kB | 93.9 kB | N/A |
| page.js gzip | 148 kB | 148 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 623 B | 626 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.5 kB | 37.5 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.07 kB | 2.07 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 01-16-Skip_postcss_config_location_resolving_in_node_modules | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 169 kB | 169 kB | ✓ |
| app-page-exp..prod.js gzip | 95.4 kB | 95.4 kB | ✓ |
| app-page-tur..prod.js gzip | 96 kB | 96 kB | ✓ |
| app-page-tur..prod.js gzip | 90.6 kB | 90.6 kB | ✓ |
| app-page.run...dev.js gzip | 142 kB | 142 kB | ✓ |
| app-page.run..prod.js gzip | 90 kB | 90 kB | ✓ |
| app-route-ex...dev.js gzip | 24.2 kB | 24.2 kB | ✓ |
| app-route-ex..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route.ru...dev.js gzip | 23.6 kB | 23.6 kB | ✓ |
| app-route.ru..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.39 kB | 9.39 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.67 kB | 9.67 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.39 kB | 9.39 kB | ✓ |
| pages-turbo...prod.js gzip | 22 kB | 22 kB | ✓ |
| pages.runtim...dev.js gzip | 22.6 kB | 22.6 kB | ✓ |
| pages.runtim..prod.js gzip | 22 kB | 22 kB | ✓ |
| server.runti..prod.js gzip | 49.6 kB | 49.6 kB | ✓ |
| Overall change | 942 kB | 942 kB | ✓ |

</details>

</details>
Commit: 9e1816a6785419ddb6f63e916afce24b388ef225

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 9e1816a6785419ddb6f63e916afce24b388ef225

`TURBOPACK=1 pnpm test-dev test/e2e/prerender.test.ts` (turbopack)
- Prerender > should handle fallback only page correctly HTML

<details>
<summary>Expand output</summary>

  ● Prerender › should handle fallback only page correctly HTML

    expect(received).toContain(expected) // indexOf

    Expected substring: "hi fallback"
    Received string:    "Post: first/post"

      849 |
      850 |       const text = await browser.elementByCss('p').text()
    > 851 |       expect(text).toContain('hi fallback')
          |                    ^
      852 |
      853 |       // wait for fallback data to load
      854 |       await check(() => browser.elementByCss('p').text(), /Post/)

      at Object.toContain (e2e/prerender.test.ts:851:20)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by sokra — APPROVED** [review]

**Pull request merged**